### DISCUSSION TITLE
time_charge_complete notification

### DISCUSSION THREAD

**Discussion by Joe123-B** [alandtse/tesla#870] [Q&A] [answered]
Hi i'm trying to include the "remaining time to complete the charge of my tesla , in a notification and have the following code in the automation

service: notify.alexa_media_bedroom_echo_dot
metadata: {}
data:
  message: >-
    Tessy the beast is charging, Estimated to complete in {{ as_timestamp( states("sensor.tessy_the_beast_time_charge_complete") ) }}'

the above gives me a Billion and some in Seconds, is there any easier way to just read the remaining time in minutes?

before trying the above code, I looked at the code of the STATE of  the time_charge_comple and found it was set to device class as below  and even showing the remaining time in minutes - but can't seem to get it to work :( 

device_class: timestamp
icon: mdi:timer-plus
friendly_name: Tessy the beast Time charge complete
minutes_to_full_charge: 200
attribution: Data provided by Tesla

**Comment by skipishere** [comment, marked as answer]
There is an attribute `minutes_to_full_charge` on that sensor which has the minutes amount. (Saves you trying to do the fun of time manipulation, the billions of seconds you are getting is the number of seconds since the epoch I believe)

**Comment by Joe123-B** [comment]
How do I use that?

**Reply by skipishere** [reply]
For your example it would be `state_attr("sensor.tessy_the_beast_time_charge_comple", "minutes_to_full_charge)`

**Comment by Joe123-B** [comment]
great thank you, any chance to get the number of hours remaining to full
charge? other than having to /60 to get the hours?

On Sat, Feb 24, 2024 at 5:47 PM Stephen Jones ***@***.***>
wrote:

> For your example it would be state_attr("sensor.tessy_the_beast_time_charge_comple",
> "minutes_to_full_charge)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alandtse/tesla/discussions/870#discussioncomment-8578148>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMTE62PTH7PPK2P2UV4XZZDYVIKQPAVCNFSM6AAAAABDYCRH56VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKNZYGE2DQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Youssef Bouanani
0033664637768

**Reply by skipishere** [reply]
Apart from the divide 60 approach, the only other option would be to take the original sensor and convert it to a timestamp take difference from the now timestamp and then use the hours from that difference value.

 Given you've got a minutes value, I'd say use that unless you want to do some clever stuff with time

**Comment by Joe123-B** [comment]
thanks a lot for your help!
yeah i've used the minutes value and divide by 60 -  but having trouble
with the decimals and also want to show/hear it  as time left :    HH hours
and MM minutes. very new to home assistant so need to figure out how to get
the time left in the appropriate format


On Sat, Feb 24, 2024 at 5:56 PM Stephen Jones ***@***.***>
wrote:

> Apart from the divide 60 approach, the only other option would be to take
> the original sensor and convert it to a timestamp take difference from the
> now timestamp and then use the hours from that difference value.
>
> Given you've got a minutes value, I'd say use that unless you want to do
> some clever stuff with time
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alandtse/tesla/discussions/870#discussioncomment-8578223>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMTE62MB5BLA5XFGVX7SRD3YVILTXAVCNFSM6AAAAABDYCRH56VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKNZYGIZDG>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Youssef Bouanani
0033664637768
